src/main/java/Duke/Commands/EventCommand.java

```diff
@@ -0,0 +1,123 @@
+package Duke.Commands;
+
+import Duke.Errors.DukeException;
+import Duke.Errors.EventException;
+import Duke.Errors.FileAbsentException;
+import Duke.Helpers.Storage;
+import Duke.Helpers.TaskList;
+import Duke.Helpers.Ui;
+import Duke.Tasks.Event;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * handles cases when Event is keyword
+ */
+public class EventCommand extends AddCommand{
+    /**
+     * Assigns string to a value of string
+     *
+     * @param string assigns string to this this.string
+     */
+    public EventCommand(String string) {
+        super(string);
+    }
+
+    private static Event provide(String name, String string, String end) throws DukeException {
+        Event e;
+        try{
+            LocalDate parsedDate = stringToLocalDate(string);
+            LocalDate endDate = stringToLocalDate(end);
+            if(parsedDate.isAfter(endDate)){
+                throw new EventException(false, false, true, false);
+            }
+            e = new Event(name, parsedDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy")),
+                    endDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy")));
+        }catch (EventException event){
+            throw new EventException(false, false, true, false);
+        }catch (DateTimeException d) {
+            try {
+                LocalDateTime parsedDate = stringToLocalDateTime(string);
+                LocalDateTime endDate = stringToLocalDateTime(end);
+                if(parsedDate.isAfter(endDate)){
+                    throw new EventException(false, false, true, false);
```

Comment:
For this part, is it better avoid nested try catch block to increase readability? Maybe you can extract each part separately into different methods :))

Reply:
Okay I will, thanks for your feedback. I agree that it is not very readable.